src/transformers/models/whisper/generation_whisper.py

```diff
@@ -438,21 +482,33 @@ def generate(
                 specific kwargs should not be prefixed and decoder specific kwargs should be prefixed with *decoder_*.
 
         Return:
-            [`~utils.ModelOutput`] or `torch.LongTensor` or `Dict[str, Any]`: A [`~utils.ModelOutput`] (if `return_dict_in_generate=True`
-            or when `config.return_dict_in_generate=True`) or a `torch.FloatTensor` or a dict of segments when `return_segments=True`.
+            [`~utils.ModelOutput`] or `Dict[str, Any]` or `torch.LongTensor`:
 
-                If the passed input is > 30 seconds / > 3000 mel input features and `return_segments=True` then a dictionary of generated sequence ids, called `sequences` and a list of each generated segment is returned.
+                A [`~utils.ModelOutput`] (when `return_dict_in_generate=True` and the passed input requires only one call to the underlying `generate` method) or a `Dict[str, Any]` (when `return_segments=True` or `return_token_timestamps=True`, with keys `sequences`, `segments` if `return_segments=True`, `token_timestamps` if `return_token_timestamps=True`;
+                `sequences` and `token_timestamps` includes the decoder input ids and end of sequence id if the passed input requires only one call to the underlying `generate` method) or a torch.LongTensor` (including the decoder input ids and end of sequence id if the passed input requires only one call to the underlying `generate` method).
 
-                else if the passed input is <= 30 seconds / >= 3000 mel input features, the possible [`~utils.ModelOutput`] types are:
+                The possible [`~utils.ModelOutput`] types are:
+                - [`~utils.GenerateEncoderDecoderOutput`]
+                - [`~utils.GenerateBeamEncoderDecoderOutput`]
 
-                    - [`~generation.GenerateEncoderDecoderOutput`],
-                    - [`~generation.GenerateBeamEncoderDecoderOutput`]
+                A passed input requires only one call to the underlying `generate` method:
+                    - necessarily if `return_timestamps=False` (and necessarily the passed input is <= 30 seconds / <= 3000 mel input features since otherwise `return_timestamps` needs to be set to `True`).
+                    - eventualy if `return_timestamps=True` and the passed input is <= 30 seconds / <= 3000 mel input features, if the last predicted timestam is not < 30 seconds which will trigger a second call to generate starting to this timestamp.
 
-                else only the generated output sequence ids are returned.
+                `segments` is a list of lists (one per batch element) of `segment`.
```

Comment:
I assume that the length of the list is equal to the number of calls to the internal `generate` method?

Reply:
No, it is a list of segments, meaning a list whose length depends on the predicted timestamp tokens. The only way to know the number of calls to the internal `generate` method here is to look at the segment's `result`.
Indeed, we can have `[{start_1, end_1, result_1}, {start_2, end_2, result_1}, {start_3, end_3, result_2}]`: the two first segments come from the same call to internal generate (result_1) and the last to the next one (result_2).